doctr/models/detection/linknet/pytorch.py

```diff
@@ -0,0 +1,275 @@
+# Copyright (C) 2021, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+from copy import deepcopy
+import torch
+from torch import nn
+from torch.nn import functional as F
+from torchvision.models._utils import IntermediateLayerGetter
+from typing import List, Dict, Any, Optional
+
+from .base import LinkNetPostProcessor, _LinkNet
+
+__all__ = ['LinkNet', 'linknet16']
+
+
+default_cfgs: Dict[str, Dict[str, Any]] = {
+    'linknet16': {
+        'layout': [64, 64, 128, 256, 512],
+        'fpn_layers': ['layer1', 'layer2', 'layer3', 'layer4'],
+        'fpn_channels': [256, 512, 1024, 2048],
+        'input_shape': (3, 1024, 1024),
+        'mean': (.5, .5, .5),
+        'std': (1., 1., 1.),
+        'rotated_bbox': False,
+        'url': None,
+    },
+}
+
+
+class LinkNetEncoder(nn.Module):
+    def __init__(self, in_chans: int, out_chans: int):
+        super().__init__()
+        self.stage1 = nn.Sequential(
+            nn.Conv2d(in_chans, out_chans, kernel_size=3, padding=1, stride=2, bias=False),
+            nn.BatchNorm2d(out_chans),
+            nn.ReLU(inplace=True),
+            nn.Conv2d(out_chans, out_chans, kernel_size=3, padding=1, bias=False),
+            nn.BatchNorm2d(out_chans),
+        )
+
+        self.shortcut = nn.Sequential(
+            nn.Conv2d(in_chans, out_chans, kernel_size=1, stride=2, bias=False),
+            nn.BatchNorm2d(out_chans),
+        ) if in_chans != out_chans else None
+
+        self.stage2 = nn.Sequential(
+            nn.Conv2d(out_chans, out_chans, kernel_size=3, padding=1, bias=False),
+            nn.BatchNorm2d(out_chans),
+            nn.ReLU(inplace=True),
+            nn.Conv2d(out_chans, out_chans, kernel_size=3, padding=1, bias=False),
+            nn.BatchNorm2d(out_chans),
+            nn.ReLU(inplace=True),
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        out = self.stage1(x)
+        if self.shortcut is not None:
+            out += self.shortcut(x)
+        out = F.relu(out, inplace=True)
+        out = self.stage2(out) + out
+
+        return out
```

Comment:
ResNet block, maybe we should code it in resnet.py and then import it here, but it can be refactored later.

Reply:
Yes I thought I would have to modify a few things first! I'll do this in a follow-up PR